### DISCUSSION TITLE
Maximum number of items in catalogue search?

### DISCUSSION THREAD

**Discussion by mennostraatsma** [microsoft/PlanetaryComputer#100] [Q&A] [answered]
I am trying to make a temporal dataset using Landsat data, but found a difference between the results on my local machine and the hub on the planetary computer. The following code selects Landsat images over three different time spans

`from pystac_client import Client

catalog = Client.open("https://planetarycomputer.microsoft.com/api/stac/v1")

aoi = {
  "type": "Polygon",
  "coordinates": [
    [
      [3.9311504572027047, 51.58324921266728],
      [6.363457484685426, 51.58324921266728],
      [6.363457484685426, 52.447153494565555],
      [3.9311504572027047, 52.447153494565555],
      [3.9311504572027047, 51.58324921266728]
    ]
  ]
}

for year in [2022, 2020, 2018]:
    search = catalog.search(
        collections=["landsat-c2-l2"], intersects=aoi,
        datetime=f"{year}-02-01/2022-08-23",
        query={"eo:cloud_cover": {"lt": 25}},
        limit=500)
    items = search.get_all_items()
    print(len(items))`

On my local machine the results are 46, 100, 100
On the hub the results are 46, 131, 241

Is there a maximum number of items that can be selected from a local machine?

**Comment by TomAugspurger** [comment, marked as answer]
What version of `pystac-client` are you using locally? There were some changes in the 0.4.x releases that affected the output (see https://github.com/stac-utils/pystac-client/issues/237 for a long discussion).

The hub is currently pinned to pystac-client 0.3.5 to avoid this issue, but the recently released pystac-client 0.5.0 fixes things (and recommends `search.item_collection()` as a replacement for `search.get_all_items()`). I'll be updating the Hub images in the next few days / week.

**Comment by mennostraatsma** [comment]
Thanks Tom
Locally, I was indeed using version 0.4.0, which is the current latest version on pip. Updating to 0.5 through conda is possible, but it will require a new environment just updating `pystac-client`  led to an inconsistent environment.
